index.js

```diff
@@ -33,26 +34,33 @@ function KoaOAuthServer(options) {
  */
 
 KoaOAuthServer.prototype.authenticate = function() {
+
   var server = this.server;
 
   return function (ctx, next) {
+
     var request = new Request(ctx.request);
     var response = new Response(ctx.response);
     var authenticate = Promise.promisify(server.authenticate, { context: server });
 
     // pass `null` for 3rd argument as NodeOAuthServer#authenticate expects callback as 4th argument
     return authenticate(request, response, null)
       .then(function (token) {
+
         ctx.state.oauth = {
           token: token
         };
+
         handleResponse.call(ctx, response);
+
+        return next();
```

Comment:
@pschwyter Hi, got a question about this code here: why are you returning `next`? Doing it like this, you are catching all exceptions and errors from downstream middleware, so if I want to propagate my downstream errors, I need to either model them like KoaOAuthServer errors, or to set `passthroughErrors = true` in which case I need to handle KoaOAuthServer errors by myself. Shouldn't it be done in that way that you pass through all non-KoaOAuthServer errors and just handle its errors with `.catch`?